src/main/scala/com/codacy/parsers/implementation/DotcoverParser.scala

```diff
@@ -3,24 +3,28 @@ package com.codacy.parsers.implementation
 import java.io.File
 
 import com.codacy.api.{CoverageFileReport, CoverageReport}
-import com.codacy.parsers.CoverageParser
-import com.codacy.parsers.util.{TextUtils, XMLoader}
+import com.codacy.parsers.util.{MathUtils, TextUtils}
+import com.codacy.parsers.{CoverageParser, XmlReportParser}
 
-import scala.util.{Failure, Success, Try}
 import scala.xml.{Elem, NodeSeq}
 
 case class StatementNode(fileIndex: Int, line: Int, covered: Boolean)
 
-object DotcoverParser extends CoverageParser {
+object DotcoverParser extends CoverageParser with XmlReportParser {
   override val name: String = "DotCover"
 
   private val RootTag = "Root"
   private val CoverageAttribute = "CoveragePercent"
   private val CoveredAttribute = "Covered"
 
-  override def parse(rootProject: File, reportFile: File): Either[String, CoverageReport] = {
-    loadXml(reportFile).map(parse(rootProject, _))
-  }
+  override def parse(rootProject: File, reportFile: File): Either[String, CoverageReport] =
+    parseReport(reportFile, s"Could not find tag <$RootTag $CoverageAttribute=...>") {
+      parse(rootProject, _)
+    }
+
+  override def validateSchema(xml: Elem): Boolean = (xml \\ RootTag \ s"@$CoverageAttribute").nonEmpty
+
+  override def getRootNode(xml: Elem): NodeSeq = xml \\ RootTag
 
   private def parse(rootProject: File, rootNode: NodeSeq): CoverageReport = {
```

Comment:
I would give another name to this function as you already have the `override def parse()`

Reply:
Done ✅ 
I looked for parse methods on every file and renamed them